lib/std/mem.zig

```diff
@@ -967,14 +967,15 @@ pub fn indexOfSentinel(comptime T: type, comptime sentinel: T, p: [*:sentinel]co
             // as we don't read into a new page. This should be the case for most architectures
             // which use paged memory, however should be confirmed before adding a new arch below.
             .aarch64, .x86, .x86_64 => if (std.simd.suggestVectorSize(T)) |block_len| {
-                comptime std.debug.assert(std.mem.page_size % block_len == 0);
                 const Block = @Vector(block_len, T);
                 const mask: Block = @splat(sentinel);
 
+                comptime std.debug.assert(std.mem.page_size % @sizeOf(Block) == 0);
+
                 // First block may be unaligned
                 const start_addr = @intFromPtr(&p[i]);
                 const offset_in_page = start_addr & (std.mem.page_size - 1);
-                if (offset_in_page < std.mem.page_size - block_len) {
+                if (offset_in_page < std.mem.page_size - @sizeOf(Block)) {
```

Comment:
While we're looking at this line, shouldn't this say `offset_in_page <= std.mem.page_size - @sizeOf(Block)`
Example values: `(4096 - 32) < (4096) - (32)` should go into the top branch.

Reply:
Good find, rebased the branch on master and fixed this.